src/v/io/cache.h

```diff
@@ -671,3 +673,23 @@ bool cache<T, Hook, Evictor, Cost>::evict_main() noexcept {
  */
 
 } // namespace experimental::io
+
+template<
+  typename T,
+  experimental::io::cache_hook T::*Hook,
+  experimental::io::cache_evictor<T> Evictor,
+  experimental::io::cache_cost<T> Cost>
+struct fmt::formatter<experimental::io::cache<T, Hook, Evictor, Cost>>
+  : fmt::formatter<std::string_view> {
+    template<typename FormatContext>
+    auto format(
+      const experimental::io::cache<T, Hook, Evictor, Cost>& cache,
+      FormatContext& ctx) const {
+        const auto stat = cache.stat();
+        return fmt::format_to(
+          ctx.out(),
+          "main {} small {}",
+          stat.main_queue_size,
+          stat.small_queue_size);
```

Comment:
This seems fine to me, but I wonder if you considered instead just having a formatter for `cache::stat`. It seems a bit more intuitive than printing the cache itself (i might expect that to show some entries in each queue for instance)

Don't feel strongly about it though